wagtail/core/query.py

```diff
@@ -237,6 +272,71 @@ def not_public(self):
         """
         return self.exclude(self.public_q())
 
+    def _get_closest_view_restriction(self, page):
+        """
+        Return the PageViewRestriction most applicable to the provided ``page`` instance.
+
+        View restrictions apply to the selected page AND their descendants, so for each
+        page, it's the restriction applied to the 'closest' ancestor that determines
+        what restrictions apply to it.
+        """
+        for vr in self.all_view_restrictions:
+            if page.id == vr.page.id or page.is_descendant_of(vr.page):
+                return vr
+
+    def for_visitor(self, user=None, include_password_protected=False):
```

Comment:
❤️ the idea.
suggestion: `for_visitor` -> `for_user`

Reply:
That was my first choice, but I felt it might be able to distinguish between 'a CMS user' and 'a user of the site'. I suppose if the docstring/accompanying documentation explains things well enough, `for_user()` might be clearer.